routes/message.js

```diff
@@ -88,6 +91,29 @@ router.ws("/",function(ws,req){
                     });
                     break
                 }
+                case READ_RECEIPT:
+                {    
+                    let msg_id;
+                    
+                    if(clients[msg.to]){   
+                        console.log(Date.now());
+                        clients[msg.to].send(JSON.stringify(msg));
+                    }
+                    Message(msg).save((err,msg)=> {
+                        if(!clients[msg.to]){
+                            if(undeliveredReadReceipts[msg.to])
```

Comment:
Use the  `undeliveredMessages` object itself

Reply:
Removed undeliveredReadReceipt Object